### ISSUE TITLE
Support of Extra Features like Character info, camera angles for shooting script

### ISSUE THREAD

**Issue by viegasfh** [piersdeseilligny/betterfountain#81]
Highland, the fountain software developed for the Mac allows one to add extra features like character info and gender analysis through json files that are saved inside the .fountain file format, which is based on TextBundle. However, people seem to overlook the versatility of the Notes feature.

One could add character biography like this:

[[CHARACTER:
NAME: James Bond
AGE: 38
PROFESSION: Spy

BIOGRAPHY: James Bond is a baddass womanizer with a gun]]

This is just an idea. Given that the notes are ignored when printing, they are just treated as comments, but it is possible to implement code that parses this information and renders it in a nice way either in the final PDF or the  preview.

In addition, the notes can be used to TAG certain words, etc... something like what Final Draft uses. We could add tags like this:

[[TAG: tagname]]

This can also be processed in the code.

These are just ideas that allow us to explore the versatility of Fountain through the Notes functionality. We can even include JSON inside the Notes and process it for certain additional features. In the rendering, we can hide this, of course. 

Something to consider.

All the best and nice job!

**Comment by piersdeseilligny** [comment]
Yep I definitely think this is a good idea, although haven't got around to working on it yet - I was also thinking of adding other features from Final Draft with this technique, like alternate dialogue, which could be something like

```
BOB
Hello
[[OR: Hi
OR: How you doing?]]
```

which would then be rendered in the editor as something like this:

![image](https://user-images.githubusercontent.com/42496508/93710870-b1a44400-fb41-11ea-86c7-2a39889d3371.png)

but obviously less ugly and more user-friendly than final draft.

For things like the character biography though, where in the fountain file do you think they should be?

**Comment by viegasfh** [comment]
Yes, that is an interesting feature as well. 

Regarding the character biography, I use sections to structure my Fountain file. Here is an example:

Title: Plot Genie Generated Story
Credit: By
Author: Plot Genie
Source: Plot Genie - Short Short Story

# The Plot Skeleton

[[CHARACTER: A duelist ]]

[[LOCALE: In a love nest]]

[[MOTIVATION: Desire to get a signal through]]

[[OBSTACLE: The threatened disloyalty of an employee]]

[[METHOD-OF-ATTACK: Offering a sacrifice to prove loyalty]]

[[CRISIS: One whose identity is mistaken is about to suffer fatal consequences]]

[[CLIMAX: A trap set for another enmeshes or threatens misfortune to a loved one of the avenger]]

# The Premise

= A duelist , one of the principal characters, experiences a situation taking place in a love nest, where someone is motivated with desire to get a signal through. The threatened disloyalty of an employee on the part of some character introduces a series of obstacles. Offering a sacrifice to prove loyalty is the approach used to extricate the obstacles. The crisis develops when one whose identity is mistaken is about to suffer fatal consequences. This is followed by a climax wherein a trap set for another enmeshes or threatens misfortune to a loved one of the avenger.

# The Characters

[[CHARACTER:
NAME: Character 1
AGE: 34
PROFESSION: Actor
BIOGRAPHY: This is character 1 biography.]]

[[CHARACTER:
NAME: Character 2
AGE: 39
PROFESSION: Dentist
BIOGRAPHY: This is character 2 biography.]]

[[CHARACTER:
NAME: Character N
AGE: 51
PROFESSION: Writer
BIOGRAPHY: This is character N biography.]]

# Plot Development

[[In this section is where we can do the plot outline for those that use it. Synopsis and Treatment as subsections.]]

# The Script

[[This is where the actual script is developed]]

FADE IN:

INT. SCENE 1 - DAY

Action goes here.

ETc....

Now, this makes everything fit into a single text file, and makes it long and it works fine if you are working with text editors. If you are working with a screenwriting software that uses Fountain at its core, I don't know if it wouldn't be easier to adopt a file format such as .fountain used in Highland 2.

I have been busing coding another project, but whenever I have time I will try some of the parsings around this. Need to play around with Typescript a little more. I use Python in most cases.

**Comment by viegasfh** [comment]
Here is an example of Shot Lists being generated from metadata inside Notes: https://slugline.co/blog/shotlists
